tutorials/networking/high_level_multiplayer.rst

```diff
@@ -371,9 +372,151 @@ have loaded the game scene.
+
+        [Signal]
+        public delegate void PlayerDisconnectedEventHandler(int peerId);
+
+        [Signal]
+        public delegate void ServerDisconnectedEventHandler();
+
+        private const int _port = 7000;
+        private const string _defaultServerIp = "127.0.0.1";
+        private const int _maxConnections = 20;
+
+        // This will contain player info for every player, with the keys being each player's unique IDs.
+        private readonly Dictionary<long, Dictionary<string, string>> _players = new();
+
+        // This is the local player info. This should be modified locally before the connection is made.
+        // It will be passed to every other peer.
+        // For example, the value of "name" can be set to something the player entered in a UI scene.
+        private readonly Dictionary<string, string> _playerInfo = new()
+        {
+            {"name", "Name"}
+        };
+
+        private int _playersLoaded = 0;
+
+        public override void _Ready()
+        {
+            Multiplayer.PeerConnected += _OnPlayerConnected;
+            Multiplayer.PeerDisconnected += _OnPlayerDisconnected;
+            Multiplayer.ConnectedToServer += _OnConnectedOk;
+            Multiplayer.ConnectionFailed += _OnConnectedFail;
+            Multiplayer.ServerDisconnected += _OnServerDisconnected;
+        }
+
+        public int JoinGame(string address = "")
+        {
+            if (string.IsNullOrEmpty(address))
+            {
+                address = _defaultServerIp;
+            }
+
+            ENetMultiplayerPeer peer = new();
+            Error error = peer.CreateClient(address, _port);
+            if (error != Error.Ok)
+            {
+                return (int)error;
+            }
+
+            Multiplayer.MultiplayerPeer = peer;
+            return (int)Error.Ok;
+        }
+
+        public int CreateGame()
+        {
+            ENetMultiplayerPeer peer = new();
+            Error error = peer.CreateServer(_port, _maxConnections);
+            if (error != Error.Ok)
+            {
+                return (int)error;
+            }
+
+            Multiplayer.MultiplayerPeer = peer;
+
+            _players[1] = _playerInfo;
+            EmitSignal(nameof(PlayerConnectedEventHandler), 1, _playerInfo);
+            return (int)Error.Ok;
+        }
+
+        public void RemoveMultiplayerPeer()
+        {
+            Multiplayer.MultiplayerPeer = null;
+        }
+
+        // When the server decides to start the game from a UI scene, do Lobby.load_game.rpc(filepath)
+        [Rpc(CallLocal = true, TransferMode = MultiplayerPeer.TransferModeEnum.Reliable)]
+        public void LoadGame(string gameScenePath)
+        {
+            GetTree().ChangeSceneToFile(gameScenePath);
+        }
+
+        // Every peer will call this when they have loaded the game scene.
+        [Rpc(mode: MultiplayerApi.RpcMode.AnyPeer, CallLocal = true, TransferMode = MultiplayerPeer.TransferModeEnum.Reliable)]
```

Comment:
Not sure if we want to use named parameters (`mode: `) for all the `Rpc` attributes.